src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -8,60 +9,95 @@ namespace NServiceBus
+                    context.AbortReceiveOperation();
+                }
+                else
+                {
+                    await MoveMessageToErrorQueue(context, fork, message, ex).ConfigureAwait(false);
+                }
+            }
+        }
 
-                    message.RevertToOriginalBodyIfNeeded();
+        async Task MoveMessageToErrorQueue(ITransportReceiveContext context, Func<IFaultContext, Task> fork, IncomingMessage message, Exception exception)
+        {
+            try
+            {
+                Logger.Error($"Moving message '{message.MessageId}' to the error queue because processing failed due to an exception:", exception);
 
-                    message.SetExceptionHeaders(exception, localAddress);
+                message.RevertToOriginalBodyIfNeeded();
 
-                    message.Headers.Remove(Headers.Retries);
+                message.SetExceptionHeaders(exception, localAddress);
 
-                    var outgoingMessage = new OutgoingMessage(message.MessageId, message.Headers, message.Body);
-                    var faultContext = this.CreateFaultContext(context, outgoingMessage, errorQueueAddress, exception);
+                message.Headers.Remove(Headers.Retries);
+                message.Headers.Remove(Headers.FLRetries);
 
-                    await fork(faultContext).ConfigureAwait(false);
+                var outgoingMessage = new OutgoingMessage(message.MessageId, message.Headers, message.Body);
+                var faultContext = this.CreateFaultContext(context, outgoingMessage, errorQueueAddress, exception);
 
-                    await context.RaiseNotification(new MessageFaulted(message, exception)).ConfigureAwait(false);
-                }
-                catch (Exception ex)
-                {
-                    criticalError.Raise("Failed to forward message to error queue", ex);
-                    throw;
-                }
+                await fork(faultContext).ConfigureAwait(false);
+
+                await context.RaiseNotification(new MessageFaulted(message, exception)).ConfigureAwait(false);
+
+                failureInfoStorage.ClearFailureInfoForMessage(message.MessageId);
+            }
+            catch (Exception ex)
+            {
+                criticalError.Raise("Failed to forward message to error queue", ex);
+
+                throw;
             }
         }
 
         CriticalError criticalError;
         string errorQueueAddress;
+        FailureInfoStorage failureInfoStorage;
         string localAddress;
+        TransportTransactionMode transportTransactionMode;
         static ILog Logger = LogManager.GetLogger<MoveFaultsToErrorQueueBehavior>();
 
         public class Registration : RegisterStep
         {
-            public Registration(string errorQueueAddress, string localAddress)
+            public Registration(string errorQueueAddress, string localAddress, TransportTransactionMode transportTransactionMode)
```

Comment:
> Can we register this the same way SecondLevelRetries is registered.

It's a feature, so isn't it a bit different? I'm not sure what you had in mind exactly, registered where?